### ISSUE TITLE
low acc for one class

### ISSUE THREAD

**Issue by ddt2014** [mahmoodlab/CLAM#138]
Thank you for your work.
I have used CLAM on my blood clot dataset, subtyping task, two class, 400/200 imgs per class, weighted sampler for training.  

During training, acc for each class (0.7/0.8 per class) and total loss (0.9) seem to be fine, but for val and test data, only one class acc is high to 0.8/0.9 and the other class acc low to 0.2/0.1.

Resnet pretrain on ImageNet and VisionTransformer trained on TCGA(from HIPT), two eature extractors provide the same result. 

After segmetation,  get 500-3000 256x256 patches for every WSI, so the bag size of each WSI same as the batch size for every model forward is 500-3000.

I think it's overfit on my dataset, but the size of dataset is almost same as your dataset.
I have reduced the model size and increase the dropout rate, the same results...

Maybe these reasons lead to my bad result:
1. the feature extractors are not suit for my data, I should finetune the extractor on my data
2. the bag size is too small, (500-3000) patches * (384/1024) vectors are not enough for CLAM model finding the differences between the two classes, I should use patch image instead of feature vector as model input, and add maybe 3/4 convs before CLAM origin model. 
3. add more data augmentation before feature extractor, instead of same feature inputs for every epoch, extract different feature for the same patch as CLAM model input 

Could you help me find the reason and the way to improve my result, thank you

**Comment by fedshyvana** [comment]
For binary problem you should look at the AUC instead of just accuracy - since the threshold is arbitrary. If AUC is high, you should be able to use the roc curve to identify an operating point (aka a different threshold than the default 0.5) to predict your classes and achieve good FPR/FNR trade offs. You can do this on the validation set for example - and evaluate on the test set using the new threshold.
